shell/packages/sandstorm-db/profile.js

```diff
@@ -18,49 +18,59 @@ var makeIdenticon;
 var httpProtocol;
 
 if (Meteor.isServer) {
+  Meteor.publish("identityProfile", function (identityId) {
+    check(identityId, String);
+
+    // Dummy query handle for the case where this.userId === identityId
+    var hasIdentityHandle = {stop: function () {} };
+
+    if (this.userId !== identityId) {
+      var hasIdentityCursor =
+          Meteor.users.find({$or: [{_id: this.userId, "loginIdentities.id": identityId},
```

Comment:
This code appears in multiple places now, right? Can we factor it out into a helper function?